app/models/piece.rb

```diff
@@ -1,3 +1,130 @@
+    y_range = []
+    if y_coord < y
+      (y_coord..y).each { |n| y_range << n }   # If statement catches if numbers go high to low
+    else
+      (y..y_coord).each { |n| y_range << n }   # Range doesn't work backwards
+      y_range = y_range.reverse               # This line reverses it back to the correct order
+    end
+    y_range.pop         # removes first number
+    y_range.shift       # removes last number
+
+    coords_to_check = []
+    y_range.each do |y|           #creates the array of coords between points
+      coords_to_check << [x,y]    
+    end
+    duplicates = coords_to_check & current_pieces_coords    # Creates a variable storing any common coords in both arrays
+    if duplicates.empty?                                    # If empty returns false, It's not obstructed
+      return false
+    else 
+      return true
+    end                             
+  end
+
+  def is_horizontally_obstructed?(x, y) 
+    x_range = []
+    if x_coord < x
+      (x_coord..x).each { |n| x_range << n }   # If statement catches if numbers go high to low
+    else
+      (x..x_coord).each { |n| x_range << n }   # Range doesn't work backwards
+      x_range = x_range.reverse               # This line reverses it back to the correct order
+    end
+    x_range.pop         # removes first number
+    x_range.shift       # removes last number
+
+    coords_to_check = []
+    x_range.each do |x|           #creates the array of coords between points
+      coords_to_check << [x,y]    
+    end
+    duplicates = coords_to_check & current_pieces_coords    # Creates a variable storing any common coords in both arrays
+    if duplicates.empty?                                    # If empty returns false, It's not obstructed
+      return false
+    else 
+      return true
+    end                             
+  end 
+
+
+  def is_diagonally_obstructed?(x, y)
+    x_range = []
+    if x_coord < x
+      (x_coord..x).each { |n| x_range << n }   # If statement catches if numbers go high to low
+    else
+      (x..x_coord).each { |n| x_range << n }   # Range doesn't work backwards
+      x_range = x_range.reverse               # This line reverses it back to the correct order
+    end
+    x_range.pop         # removes first number
+    x_range.shift       # removes last number
+                                  
+    y_range = []
+    if y_coord < y 
+      (y_coord..y).each { |n| y_range << n }
+    else
+      (y..y_coord).each { |n| y_range << n }
+      y_range = y_range.reverse
+    end
+    y_range.pop
+    y_range.shift
+
+    coords_to_check = x_range.zip(y_range)   # Combines x and y ranges to make each coord
+    duplicates = coords_to_check & current_pieces_coords    # Creates a variable storing any common coords in both arrays
+    if duplicates.empty?                                    # If empty returns false, It's not obstructed
+      return false
+    else 
+      return true
+    end
+  end
+
+
+  def invalid_input?(x, y)
```

Comment:
Input is invalid if the direction is not diagonal, vertical or horizontal, right? Is there a way we can shorten the method quite a bit? 😄